shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -1230,9 +1230,10 @@ object GlutenConfig {
   val COLUMNAR_VELOX_CONNECTOR_IO_THREADS =
     buildStaticConf("spark.gluten.sql.columnar.backend.velox.IOThreads")
       .internal()
-      .doc("The IO threads for connector split preloading")
+      .doc("The Size of the IO thread pool in the Connector. This thread pool is used for split" +
+        " preloading and DirectBufferedInput.")
       .intConf
-      .createWithDefault(0)
+      .createWithDefaultFunction(() => SQLConf.get.getConfString("spark.executor.cores", "1").toInt)
```

Comment:
Do you mean if the executor cores is 4, then there are at most 4 * 4 IO Threads at native side in one executor?

Reply:
Or the IOThread is shared by all tasks in one executor ?

Reply:
It's shared by all tasks in one exectuor.

Reply:
Does the IO Threads is related to the task num in one executor? The task num in one executor is determined by `spark.executor.cores`/`spark.task.cpus`, so we may get 8 IO Threads while there are only 4 tasks running in one executor.

Reply:
Make sense, updated.